scripts/load_mentors.js

```diff
@@ -143,10 +160,49 @@ exports.handler = async (event) => {
   for (const mentorsChunk of chunks(indexMentors, ELASTIC_CHUNK_SIZE)) {
     i += 1;
     console.log(`Indexing mentors (chunk ${i})...`);
-    const indexBody = mentorsChunk.flatMap(([id, mentor]) => [
-      { index: { _index: ELASTIC_INDEX, _id: id } },
-      mentor,
-    ]);
+    const indexBody = (
+      await Promise.all(
+        mentorsChunk.map(async ([id, mentor]) => {
+          const newMentorObject = { ...mentor };
+          await Promise.all(
+            [
+              [mentor.full_image_url, `${id}-full`, "full_image_url"],
+              [
+                mentor.thumbnail_image_url,
+                `${id}-thumbnail`,
+                "thumbnail_image_url",
+              ],
+            ].map(async ([imageURL, imageKey, objectProperty]) => {
+              if (imageURL === PLACEHOLDER_THUMBNAIL_URL) {
+                s3.deleteObject(
+                  {
+                    Bucket: AWS_S3_BUCKET_NAME,
+                    Key: `${AWS_S3_IMAGE_FOLDER}${imageKey}.jpg`,
+                  },
+                  function (err, data) {}
+                );
+                return [{ index: { _index: ELASTIC_INDEX, _id: id } }, mentor];
+              }
+              const res = await fetch(imageURL);
+              const blob = await res.buffer();
```

Comment:
It seems that the code assumes the image file is `.jpg`. Could this part be rewritten to detect the file extension (e.g. using [`extname`](https://nodejs.dev/en/learn/nodejs-file-paths/)), with `imageKey` containing the file extension?

Reply:
Oh about this, actually the `extname` is null on the Airtable side (the image url is something like [this](https://v5.airtableusercontent.com/v1/14/14/1673258400000/mw_ERQvhV3jAjaoG4fimgw/DW1FtB48Gy2_gEEktGNdOdeQCSaZ9xXMi2P4_9WTixXq-0QydHnwfE1lpys1IqFVJwVHU7L6TooC25qhrbE06g/AVtS73segRN0AunhWWvYYmyTNnFJtX4cQIDD5CSRaN4)). Airtable's response has a Content-Type header showing `image/jpeg`, such that the contents are rendered as a jpeg image

For now, I used an approach to more closely follow Airtable's URLs, by **not** putting an extension, and instead copying the Content-Type from the Airtable response. This way if Airtable suddenly decides to upload a png image, the content type in the response by S3 would also match it.

Another possible alternative is changing the extension based on the content-type, but I think that may be more complicated. When deleting the images in the lambda script, we would need to identify what extension the uploaded image already has, which could add unnecessary logic